crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -297,102 +305,212 @@ impl TransactionBlock {
+
+                let mut bound = TxLookupBound::from_range((Some(lo_tx), Some(hi_tx)));
+
+                // If `transaction_ids` is specified, we can use that in lieu of the range under the
+                // assumption that it will further constrain the rows we look up.
+                if let Some(txs) = &filter.transaction_ids {
+                    let transaction_ids: Vec<TxLookup> = conn.results(move || select_ids(txs, &bound).into_boxed())?;
+                    if transaction_ids.is_empty() {
+                        return Ok::<_, diesel::result::Error>((false, false, vec![]));
+                    }
+                    bound = TxLookupBound::from_set(transaction_ids.into_iter().map(|tx| tx.tx_sequence_number as u64).collect());
+                }
+
+                let mut subqueries = vec![];
+
+                if let Some(f) = &filter.function {
+                    subqueries.push(match f {
+                        FqNameFilter::ByModule(filter) => match filter {
+                            ModuleFilter::ByPackage(p) => {
+                                (select_pkg(p, sender, &bound), "tx_calls_pkg")
+                            }
+                            ModuleFilter::ByModule(p, m) => {
+                                (select_mod(p, m.clone(), sender, &bound), "tx_calls_mod")
+                            }
+                        },
+                        FqNameFilter::ByFqName(p, m, n) => (
+                            select_fun(p, m.clone(), n.clone(), sender, &bound),
+                            "tx_calls_fun",
+                        ),
+                    });
+                }
+                if let Some(kind) = &filter.kind {
+                    subqueries.push((select_kind(*kind, &bound), "tx_kinds"));
+                    if let Some(sender) = &sender {
+                        subqueries.push((select_sender(sender, &bound), "tx_senders"));
+                    }
+                }
+                if let Some(recv) = &filter.recv_address {
+                    subqueries.push((select_recipient(recv, sender, &bound), "tx_recipients"));
+                }
+                if let Some(input) = &filter.input_object {
+                    subqueries.push((select_input(input, sender, &bound), "tx_input_objects"));
+                }
+                if let Some(changed) = &filter.changed_object {
+                    subqueries.push((
+                        select_changed(changed, sender, &bound),
+                        "tx_changed_objects",
+                    ));
+                }
+
+                if filter.num_filters() == 0 {
+                    // We don't need to explicitly specify sender unless it's the only filter
+                    if let Some(sender) = &sender {
+                        subqueries.push((select_sender(sender, &bound), "tx_senders"));
+                    }
+                    // And if there are no filters at all, we can operate directly on the main table
```

Comment:
If there are no additional filters, then we perform no query at all -- we can calculate the the range of transaction sequence numbers to multi-get purely based on the bounds we have been given.